src/Microsoft.TestPlatform.Client/DesignMode/DesignModeTestHostLauncherFactory.cs

```diff
@@ -13,21 +13,9 @@ public static class DesignModeTestHostLauncherFactory
     {
         private static ITestHostLauncher defaultLauncher;
 
-        private static ITestHostLauncher debugLauncher;
-
         public static ITestHostLauncher GetCustomHostLauncherForTestRun(IDesignModeClient designModeClient, TestRunRequestPayload testRunRequestPayload)
         {
-            ITestHostLauncher testHostLauncher = null;
-            if (!testRunRequestPayload.DebuggingEnabled)
-            {
-                testHostLauncher = defaultLauncher = defaultLauncher ?? new DesignModeTestHostLauncher(designModeClient);
-            }
-            else
-            {
-                testHostLauncher = debugLauncher = debugLauncher ?? new DesignModeDebugTestHostLauncher(designModeClient);
-            }
-
-            return testHostLauncher;
+            return (defaultLauncher = defaultLauncher ?? new DesignModeTestHostLauncher(designModeClient));
```

Comment:
This is unnecessarily clever. Consider assigning the value and and returning it on different lines, and using this. to prefix the field.

Reply:
Done. Reverted to original.